PalasoUIWindowsForms.Tests/SettingsProviderTests.cs

```diff
@@ -0,0 +1,114 @@
+			for (var which = 0; which < 1; which++)
+			{
+				using (var tempFolder = new TemporaryFolder())
+				{
+					var firstDirPath = Path.Combine(tempFolder.Path, "first");
+					var secondDirPath = Path.Combine(tempFolder.Path, "second");
+					var firstConfigFile = Path.Combine(firstDirPath, CrossPlatformSettingsProvider.UserConfigFileName);
+					var secondConfigFile = Path.Combine(firstDirPath, CrossPlatformSettingsProvider.UserConfigFileName);
+					Directory.CreateDirectory(firstDirPath);
+					Directory.CreateDirectory(secondDirPath);
+					if (which == 0)
+						File.WriteAllText(firstConfigFile, @"nonsense");
+					else
+						File.WriteAllText(secondConfigFile, @"nonsense");
+
+					var list = new List<string>();
+					list.Add(secondDirPath);
+					list.Add(firstDirPath);
+					list.Sort(CrossPlatformSettingsProvider.VersionDirectoryComparison);
+					if (which == 0)
+						Assert.That(list[0], Is.EqualTo(firstDirPath), "first directory has config so should have come first");
+					else
+						Assert.That(list[0], Is.EqualTo(secondDirPath), "second directory has config so should have come first");
+				}
+			}
+		}
+
+		[Test]
+		public void SettingsFolderWithNewerConfig_SortsBeforeOneWithOlderConfig()
+		{
+
+			using (var tempFolder = new TemporaryFolder())
+			{
+				var firstDirPath = Path.Combine(tempFolder.Path, "first");
+				var secondDirPath = Path.Combine(tempFolder.Path, "second");
+				var firstConfigFile = Path.Combine(firstDirPath, CrossPlatformSettingsProvider.UserConfigFileName);
+				var secondConfigFile = Path.Combine(secondDirPath, CrossPlatformSettingsProvider.UserConfigFileName);
+				Directory.CreateDirectory(firstDirPath);
+				Directory.CreateDirectory(secondDirPath);
+				File.WriteAllText(firstConfigFile, @"nonsense");
+				File.WriteAllText(secondConfigFile, @"nonsense"); // second is newer
+
+				var result = CrossPlatformSettingsProvider.VersionDirectoryComparison(firstDirPath, secondDirPath);
+				Assert.That(result, Is.GreaterThan(0));
+
+				File.WriteAllText(firstConfigFile, @"nonsense"); // now first is newer
+				result = CrossPlatformSettingsProvider.VersionDirectoryComparison(firstDirPath, secondDirPath);
+				Assert.That(result, Is.LessThan(0));
+
+				// A final check to make sure it is really working the way we want
+				var list = new List<string>();
+				list.Add(secondDirPath);
+				list.Add(firstDirPath);
+				list.Sort(CrossPlatformSettingsProvider.VersionDirectoryComparison);
+				Assert.That(list[0], Is.EqualTo(firstDirPath));
+			}
+		}
+
+		/// <summary>
+		/// This test was primarily created to test that we can override the default location for the RegistrationSettingsProvider.
+		/// However it's also quite a useful test of the whole business of reading settings.
+		/// </summary>
+		[Test]
+		public void CanOverrideDefaultLocation()
+		{
+			RegistrationSettingsProvider.SetProductName("BloomUnitTest");
```

Comment:
I know Bloom was the whole point of this, and I'm not overly bothered by this, but if you have to update this file anyhow you might change the name to 'FlowerBeta' or something less obviously focused on a single product. (Keeping product names out of even the tests and comments is helpful to avoid leading junior developers into the trap of putting product specific code into a generic library)